lib/TPP/ToBlockLayoutAndBack.cpp

```diff
@@ -68,7 +68,10 @@ static Value toUnPackLayoutImpl(OpBuilder &builder, Location loc, Value input,
 static Value handleLayoutNC_NCnc(OpBuilder &builder, Location loc, Value input,
                                  Value output, ArrayRef<OpFoldResult> tiles) {
   assert(tiles.size() == 2 && "expect two tile sizes for NC_NCnc");
+  int64_t rank = input.getType().cast<RankedTensorType>().getRank();
   SmallVector<int64_t> innerDimPos = {0, 1};
+  if (rank == 3 || rank == 5)
```

Comment:
Magic numbers. What does this mean?

Reply:
yes, not really great. Here we need to deal with batchMatmul. For now we have this layout only for matmul and batch matmul, the if stmt sets the right inner dims pos and outer dim perm for the batch matmul.